### ISSUE TITLE
Felix logging is unhelpful [no further additions to this now please!]

### ISSUE THREAD

**Issue by matthewdupre** [Metaswitch/calico#363]
Please add small Felix logging problems to this issue.

**Comment by matthewdupre** [comment]
"[WARNING][3] calico.felix.fetcd 272: Global config changed but we don't yet support dynamic config key=/calico/config/InterfacePrefix" error looks interesting, but doesn't tell you the new / old values of the key.  This log can appear frequently, and without knowing this it's impossible to tell if it's harmless.

**Comment by matthewdupre** [comment]
`2015-04-15 12:56:06,643 [ERROR][5] calico.felix.fiptables 434: Failed to run iptables-restore.
Output:

Error:
"iptables-restore: line 2 failed

Input was:
*filter
--delete INPUT --jump felix-INPUT
--insert INPUT --jump felix-INPUT
COMMIT

2015-04-15 12:56:06,644 [ERROR][5] calico.felix.fiptables 436: Non-retryable error on line 2: '--delete INPUT --jump felix-INPUT'
2015-04-15 12:56:06,644 [DEBUG][5] calico.felix.fiptables 194: Failed to do atomic delete/insert, assuming rule wasn't programmed.`

Later followed by 

`2015-04-15 12:56:06,669 [DEBUG][5] calico.felix.fiptables 194: Failed to do atomic delete/insert, assuming rule wasn't programmed.`

and a successful program.

These ERROR logs are much too scary for something that appears to be harmless and expected where the tables aren't already set up.

**Comment by matthewdupre** [comment]
There's no log for creation of (say), an ActiveIpset.  All actors should log on creation.

**Comment by fasaxc** [comment]
It's very hard to tell what was going on with actor life cycles; pull req to improve that: #380

**Comment by fasaxc** [comment]
When we fail to connect to etcd at start of day, we make a lot of scary logs:

```
Apr 18 20:51:28.829980 10.41.0.21 err calico-felix: Failed to retrieve ready flag from etcd, waiting...
Traceback (most recent call last):
  File "/usr/lib64/python2.7/site-packages/calico/felix/fetcd.py", line 91, in wait_for_ready
  File "/usr/lib64/python2.7/site-packages/etcd/client.py", line 348, in read
  File "/usr/lib64/python2.7/site-packages/etcd/client.py", line 585, in api_execute
  File "/usr/lib64/python2.7/site-packages/etcd/client.py", line 540, in _next_server
EtcdException: No more machines in the cluster
```

Piston Cloud asked if it'd make sense to suppress the stack trace here since it's an expected scenario that we fail to connect.

**Comment by fasaxc** [comment]
Hard to diagnose incorrect felixhostname config, was #364

**Comment by fasaxc** [comment]
Spammy errors due to interface disappearing before we program it.  Should they be warnings?

```
calicotest.openstack.basetest: ERROR: calico.felix.devices 211: Cannot check flags for interface tape4d4cb49-ae (/sys/class/net/tape4d4cb49-ae/flags) - assume down
Traceback (most recent call last):
File "/usr/lib/python2.7/dist-packages/calico/felix/devices.py", line 204, in interface_up
flags = f.read().strip()
IOError: [Errno 19] No such device
```

**Comment by fasaxc** [comment]
More of the same:

```
calicotest.openstack.basetest: ERROR: calico.felix.endpoint 361: Failed to configure interface tapfc5b6c21-8e for fc5b6c21-8eb5-4e28-9bee-d1901d1806e1
Traceback (most recent call last):
File "/usr/lib/python2.7/dist-packages/calico/felix/endpoint.py", line 349, in _configure_interface
self.endpoint["mac"])
File "/usr/lib/python2.7/dist-packages/calico/felix/devices.py", line 186, in set_routes
add_route(ip_type, ip, interface, mac)
File "/usr/lib/python2.7/dist-packages/calico/felix/devices.py", line 149, in add_route
futils.check_call(["ip", "route", "replace", ip, "dev", interface])
File "/usr/lib/python2.7/dist-packages/calico/felix/futils.py", line 99, in check_call
args, retcode, stdout, stderr)
FailedSystemCall: Failed system call (retcode : 2, args : ('ip', 'route', 'replace', u'10.28.2.131', 'dev', u'tapfc5b6c21-8e'))
stdout  :
stderr  : RTNETLINK answers: Network is down
```

**Comment by neiljerram** [comment]
This is being tackled for 0.18, but that won't be possible if lots of further stuff gets dumped into here, so please no more of that.  Put any new spammy log reports into new issues.

**Comment by fasaxc** [comment]
I believe these are all addressed by #492.
